src/app/pages/common/entity/table/table.component.ts

```diff
@@ -199,6 +216,24 @@ export class TableComponent implements OnInit, AfterViewInit, AfterViewChecked {
     this.tableService.getData(this);
   }
 
+  updateColumns(): void {
+    this.displayedColumns = this._tableConf.columns
+      .map((column) => {
+        if (this.dataSource && column?.hiddenIfEmpty) {
+          const hasSomeData = this.dataSource.some((row) => row[column.prop]?.toString().trim());
+          column.hidden = !hasSomeData;
```

Comment:
This will reset `hidden` to `false` if it was set to `true` originally if `hasSomeData` is false (meaning the column is empty). So, if the intention was to hide this column regardless if it was empty or not, it will get set to `false` if it's not empty.
It should be `column.hidden = column.hidden || !hasSomeData` or something like that

Reply:
Oh sh... I'll fix it 👍🏻

Reply:
Updated